addons/purchase_stock/models/product.py

```diff
@@ -12,8 +13,16 @@ class ProductTemplate(models.Model):
     @api.model
     def _get_buy_route(self):
         buy_route = self.env.ref('purchase_stock.route_warehouse0_buy', raise_if_not_found=False)
-        if buy_route:
-            return buy_route.ids
+        try:
+            if buy_route:
+                buy_route.check_access_rule('read')
+                return buy_route.ids
+        except AccessError:
+            routes = self.env['stock.location.route'].search([('product_selectable', '=', True)])
+            for r in routes:
+                for rule in r.rule_ids:
+                    if rule.action == 'buy':
+                        return r.ids
```

Comment:
```suggestion
        if buy_route.exists():
            return buy_route.ids
```

Reply:
in fact ref is already doing the exist stuff. Maybe just add do a search instead

Reply:
Not sure if only doing the search will work as it makes a lot of tests fail without any apparent reason (like TestPurchaseOrder.test_pol_description)
I'll have a deeper look next week

Reply:
@amoyaux Seems to work if we order the routes by id, what do you think ?

Reply:
I mean to do
```python
buy_route_id = self['ir.model.data'].xmlid_to_object('purchase_stock.route_warehouse0_buy', raise_if_not_found=False)
buy_route = self.env.search([('id', '=', buy_route_id)])
...
```